tests/wind_data_test.py

```diff
@@ -0,0 +1,264 @@
+    assert wind_rose.n_findex == 6
+
+
+def test_unpack_for_reinitialize():
+    wind_directions = np.array([270, 280, 290])
+    wind_speeds = np.array([6, 7])
+    freq_table = np.array([[1.0, 0.0], [0, 1.0], [0, 0]])
+
+    # First test using default assumption only non-zero frequency cases computed
+    wind_rose = WindRose(wind_directions, wind_speeds, freq_table)
+
+    (
+        wind_directions_unpack,
+        wind_speeds_unpack,
+        ti_table_unpack,
+    ) = wind_rose.unpack_for_reinitialize()
+
+    # Given the above frequency table, would only expect the
+    # (270 deg, 6 m/s) and (280 deg, 7 m/s) rows
+    np.testing.assert_allclose(wind_directions_unpack, [270, 280])
+    np.testing.assert_allclose(wind_speeds_unpack, [6, 7])
+
+
+def test_wind_rose_resample():
+    wind_directions = np.array([0, 2, 4, 6, 8, 10])
+    wind_speeds = np.array([8])
+    freq_table = np.array([[1.0], [1.0], [1.0], [1.0], [1.0], [1.0]])
+
+    wind_rose = WindRose(wind_directions, wind_speeds, freq_table)
+
+    # Test that resampling with a new step size returns the same
+    wind_rose_resample = wind_rose.resample_wind_rose()
+
+    np.testing.assert_allclose(wind_rose.wind_directions, wind_rose_resample.wind_directions)
+    np.testing.assert_allclose(wind_rose.wind_speeds, wind_rose_resample.wind_speeds)
+    np.testing.assert_allclose(wind_rose.freq_table_flat, wind_rose_resample.freq_table_flat)
+
+    # Now test resampling the wind direction to 5 deg bins
+    wind_rose_resample = wind_rose.resample_wind_rose(wd_step=5.0)
+    np.testing.assert_allclose(wind_rose_resample.wind_directions, [0, 5, 10])
+    np.testing.assert_allclose(wind_rose_resample.freq_table_flat, [2 / 6, 2 / 6, 2 / 6])
+
+
+def test_wrap_wind_directions_near_360():
+    wd_step = 5.0
+    wd_values = np.array([0, 180, 357, 357.5, 358])
+    time_series = TimeSeries(np.array([0]), np.array([0]))
+
+    wd_wrapped = time_series._wrap_wind_directions_near_360(wd_values, wd_step)
+
+    expected_result = np.array([0, 180, 357, -wd_step / 2.0, -2.0])
+    assert np.allclose(wd_wrapped, expected_result)
+
+
+def test_time_series_to_wind_rose():
+    # Test just 1 wind speed
+    wind_directions = np.array([259.8, 260.2, 264.3])
+    wind_speeds = np.array([5.0, 5.0, 5.1])
+    time_series = TimeSeries(wind_directions, wind_speeds)
+    wind_rose = time_series.to_wind_rose(wd_step=2.0, ws_step=1.0)
+
+    # The wind directions should be 260, 262 and 264 because they're binned
+    # to the nearest 2 deg increment
+    assert np.allclose(wind_rose.wind_directions, [260, 262, 264])
+
+    # Freq table should have dimension of 3 wd x 1 ws because the wind speeds
+    # are all binned to the same value given the `ws_step` size
+    freq_table = wind_rose.freq_table
+    assert freq_table.shape[0] == 3
+    assert freq_table.shape[1] == 1
+
+    # The frequencies should [2/3, 0, 1/3]
+    # TODO this one I dont follow, @paul or @misha?
```

Comment:
@paulf81 @misi9170 could one of you add some detail here? Why should the frequencies be 2/3, 0, 1/3

Reply:
how is this?

Reply:
That's great, thank you